src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -814,6 +814,9 @@ void java_bytecode_languaget::methods_provided(id_sett &methods) const
   // Add all concrete methods to map
   for(const auto &kv : method_bytecode)
     methods.insert(kv.first);
+  // Add all synthetic methods to map
```

Comment:
What is a "synthetic method"? Does this in any way relate to stubbing? If so, could we please use a consistent terminology across the code base?

Reply:
I think synthetic methods are things like `clinit_wrapper` that we introduce to handle some JVM mechanisms, I could be wrong though? Perhaps could be defined in, e.g., `synthetic_methods_map.h`?

Reply:
Sort of. The synthetic methods are either clinit methods, generated for stub types, or static initializer method wrappers, generated for all static initializers to ensure they only run once, generated whether a stub type or not. In future the methods defined for array types (e.g. `.clone()`) will probably go in this map too. Stub bodies are currently provided with a different mechanism but could use this too.

In general a synthetic method is one generated internally by the language frontend without corresponding user bytecode. Stubs are thus synthetic, but not all synthetics are stubs.

Reply:
I suggest this definition goes somewhere find able in the codebase.

Reply:
This isn't really related to this PR, so I've made https://github.com/diffblue/cbmc/pull/1953